### DISCUSSION TITLE
pfkey unable to add SAD entry with SPI c27dfe6f: Function not implemented (38)

### DISCUSSION THREAD

**Discussion by ygm521** [strongswan/strongswan#1021] [General]
- config
use esp_proposals = aes128gcm128-x25519

- error
selected proposal: ESP:AES_GCM_16_128/CURVE_25519/NO_EXT_SEQ
pfkey unable to add SAD entry with SPI c27dfe6f: Function not implemented (38)
pfkey unable to add SAD entry with SPI c26c4e5e: Function not implemented (38)
unable to install inbound and outbound IPsec SA (SAD) in kernel

**Comment by tobiasbrunner** [comment]
Your kernel (interface) apparently doesn't support AES-GCM. If you did this on Linux, that's expected as Linux doesn't support AEAD algorithms via PF_KEY interface. One of the reasons why you don't want to use the _kernel-pfkey_ plugin there, but _kernel-netlink_ instead.
